repository/src/main/java/org/apache/atlas/tasks/AtlasTaskService.java

```diff
@@ -124,6 +133,197 @@ public void retryTask(String taskGuid) throws AtlasBaseException {
+                    supportedTypes.addAll(supportedTaskTypes);
+                } catch (NoSuchFieldException | IllegalAccessException e) {
+                    LOG.error("Error occurred while accessing field: ", e);
+                    throw new AtlasBaseException("Error occurred while accessing field", e);
+                }
+            }
+        }
+        List<AtlasTask> createdTasks = new ArrayList<>();
+
+        if (CollectionUtils.isNotEmpty(tasks)) {
+            for (AtlasTask task : tasks) {
+                String taskType = task.getType();
+                if (!supportedTypes.contains(taskType)) {
+                    throw new AtlasBaseException(AtlasErrorCode.TASK_TYPE_NOT_SUPPORTED, task.getType());
+                }
+                if (isClassificationTaskType(taskType)) {
+                    String classificationName = task.getClassificationName();
+                    String entityGuid = task.getEntityGuid();
+                    String classificationId = resolveAndReturnClassificationId(classificationName, entityGuid);
+                    if (StringUtils.isEmpty(classificationId)) {
+                        throw new AtlasBaseException(AtlasErrorCode.TASK_INVALID_PARAMETERS, task.toString());
+                    }
+                    task.getParameters().put(PARAM_CLASSIFICATION_VERTEX_ID, classificationId);
+                }
+                task.setUpdatedTime(new Date());
+                task.setCreatedTime(new Date());
+                task.setStatusPending();
+                task.setAttemptCount(0);
+                task.setGuid(UUID.randomUUID().toString());
+                task.setCreatedBy(RequestContext.getCurrentUser());
+
+                createTaskVertex(task);
+                createdTasks.add(task);
+            }
+        }
+
+        graph.commit();
+        RequestContext.get().endMetricRecord(metric);
+        return createdTasks;
+    }
+
+    private boolean isClassificationTaskType(String taskType) {
+         return ClassificationPropagateTaskFactory.supportedTypes.contains(taskType);
+    }
+
+    private String resolveAndReturnClassificationId(String classificationName, String entityGuid) throws AtlasBaseException {
+        String ret = null;
+        AtlasVertex entityVertex = AtlasGraphUtilsV2.findByGuid(entityGuid);
+        if (entityVertex == null) {
+            throw new AtlasBaseException(AtlasErrorCode.INSTANCE_GUID_NOT_FOUND, entityGuid);
+        }
+
+        AtlasVertex classificationVertex = GraphHelper.getClassificationVertex(entityVertex, classificationName);
+
+        if (classificationVertex != null) {
+            ret = classificationVertex.getIdForDisplay();
+        }
+
+        return ret;
+    }
+
+    @Override
+    public List<AtlasTask> deleteAtlasTasks(List<AtlasTask> tasks) {
+        AtlasPerfMetrics.MetricRecorder metric = RequestContext.get().startMetricRecord("deleteAtlasTasks");
```

Comment:
Move metric recording to [TaskREST.java]

Reply:
Done

Reply:
Done